icon: update docs

atlee025 opened the pull request:
Icon document was lacking detail information and guidance on usage. The documents for icon have been updated to include guidance and component usage.

[View preview](https://design-system.agriculture.gov.au/pr-preview/pr-1997)

[View icon documentation preview](https://design-system.agriculture.gov.au/pr-preview/pr-1997/components/icon)

## Checklist

**Preflight**

- [x] Prefix the PR title with the slug of the package or component - e.g. `accordion: Updated padding` or `docs: Updated header links`
- [x] Describe the changes clearly in the PR description
- [x] Read and check your code before tagging someone for review
- [x] Create a changeset file by running `yarn changeset`. [Learn more about change management](https://design-system.agriculture.gov.au/guides/change-management).

**Documentation**

- [x] Create or update documentation on the website
- [ ] Create or update stories for Storybook
- [ ] Create or update stories for Playroom snippets

stowball commented on packages/react/src/icon/docs/overview.mdx:
````diff
@@ -1,14 +1,65 @@
 ```
 
-[See the full list of available icons](/foundations/icons).
+Meaningful icons can be used to enhance the experience for sighted users making it easier to scan content, identify actions and understand user interfaces. The icon component gives you access to a suite of system icons that are designed to represent common states and universal actions.
````
```suggestion
Meaningful icons can be used to enhance the experience for sighted users; making it easier to scan content, identify actions and understand user interfaces. The icon component gives you access to a suite of system icons that are designed to represent common states and universal actions.
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,65 @@
+- Use icons to support visual recognition of repeatable or consistent system wide states or actions
+- Always use a meaningful text label and accessible name when using an icon in an action
+- Avoid using bespoke icons in your applications as you may confuse users
+- Only consider using an icon without a meaningful label if there is insufficient space and including the label will detract from the user experience.
```
```suggestion
- Only consider using an icon without a visible label if there is insufficient space and including the label will detract from the user experience.
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,65 @@
+- Always use a meaningful text label and accessible name when using an icon in an action
+- Avoid using bespoke icons in your applications as you may confuse users
+- Only consider using an icon without a meaningful label if there is insufficient space and including the label will detract from the user experience.
+  - An example of this is mobile search bar where including a label for the action will take up essential space for search input, and the icon meaning is explicit like ’search’ or ’close’.
```
This should not be a root bullet point

stowball replied:
```suggestion
  - An example of this is a mobile search bar. Including a label for the action will take up essential space needed by the search input, and the context provides clear meaning to the icon.
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,65 @@
+
+## Icons
+
+When using an icon, always provide a meaningful label and accessible name to help all users quickly recognise and understand what action will occur. Don’t use icons without labels because you may confuse some users - even the simplest icons will be new or uncommon to some users.
```
"Don’t use icons without labels because you may confuse some users" This kinda contradicts everything which has been said before…

atlee025 replied:
Updated and moved to the new heading `Icons and labels`. The following paragraphs have guidance on icons without labels.

"Provide a meaningful label and accessible name with the icon to help all users quickly recognise and understand what action will occur. Icons are small visual elements that can be interpreted differently by people. Using icons without labels can confuse some users - even the simplest icons will be new or uncommon to some users."

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,65 @@
+
+Each icon in our library is meaningful and universal, to ensure users understand them. Icons should not be too detailed, to ensure they’re clear and legible at all available sizes. [See the full list of available icons.](/foundations/icons)
+
+Use icons that accurately communicate the action with a label that describes the action. For example, search fields on mobile often use a magnifying glass icon without a label due to space constraints.
```
This is one weird, long sentence

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,65 @@
+
+Icons are available in sizes medium, large and extra large.
+
+Medium icons are used in small buttons, in components like accordions, drop down menus, breadcrumbs and links that are nested in content paragraphs such as external link. Always consider the complexity of an icon before using the small variant as small complex icons may be hard to read for some users.
```
```suggestion
Medium icons are used in small buttons, in components like accordions, drop-down menus, breadcrumbs and links that are nested in content paragraphs. Always consider the complexity of an icon before using the small variant, as small complex icons may be hard to read for some users.
```

adrian-yee commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,65 @@
+
+Each icon in our library is meaningful and universal, to ensure users understand them. Icons should not be too detailed, to ensure they’re clear and legible at all available sizes. [See the full list of available icons.](/foundations/icons)
+
+Use icons that accurately communicate the action with a label that describes the action. For example, search fields on mobile often use a magnifying glass icon without a label due to space constraints.
```
```suggestion
Use icons and labels together that that are commonly used and accurately communicate the action, context or outcome. For example, [Searchbox](/components/searchbox) uses a magnifying glass icon with the label 'Search' in situations where there is enough space for both the icon and label. On mobile screens the [Searchbox](/components/searchbox) responds to a smaller component where only the common and meaningful magnifying glass icon is provided. 
```

atlee025 replied:
Search box doesn't support using both icon and text. I've taken a shot at reword the example using `toggle-button`.

"Use icons and labels together that are commonly used and accurately communicate an action, context or outcome. For example, a flag [Toggle button](/components/toggle-button) can be used without a visible label provided there is enough context surrounding the icon, such as in a mailbox application to action its importance or required attention. [Toggle button](/components/toggle-button) will provide an accessible label for assistive technologies."

adrian-yee commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,65 @@
+
+<DoHeading />
+
+- Only use universal icons, that all users understand, and that clearly represent an intended meaning or context
```
```suggestion
- Use icons, that all users understand, and that clearly represent an intended meaning or context.
```

adrian-yee commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,65 @@
 ---
 title: Icon
-description: The Agriculture Design System supports the use of universal system icons. The icon component is used to apply our set of universal icons to more complex components of the system.
+description: The Agriculture Design System supports the use of meaningful and universal system icons that users understand. The icon component is used to apply our set of universal icons to more complex components of the system.
```
```suggestion
description: Icons that are meaningful and understood by all users can improve a user's experience. The icon component is used to apply our set of meaningful icons to more complex components of the system.
```

adrian-yee reviewed (commented):
Hi Alex, I tried to remove the term universal as we received negative feedback about it in recent a11y audit. I have also tried to reduce repetition a bit

adrian-yee commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,65 @@
+- Use icons to support visual recognition of repeatable or consistent system wide states or actions
+- Always use a meaningful text label and accessible name when using an icon in an action
+- Avoid using bespoke icons in your applications as you may confuse users
+- Only consider using an icon without a meaningful label if there is insufficient space and including the label will detract from the user experience.
```
we dont use second level dot points elsewhere - is there another way to handle this content?

stowball replied:
We actually do in Button. It's weird though https://design-system.agriculture.gov.au/components/button

atlee025 replied:
Moved the 2nd level dot point into an `Icons and labels` section in the 2nd paragraph.

adrian-yee reviewed (commented):
sorry I missed this comment last time

atlee025 commented:
Moved some paragraphs around and added an `Icon and labels` section that speaks on uses/guidance/explanations of labels.

[Link to docs](https://design-system.agriculture.gov.au/pr-preview/pr-1997/components/icon#icons)

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,6 +1,6 @@
 ---
 title: Icon
-description: The Agriculture Design System supports the use of meaningful and universal system icons that users understand. The icon component is used to apply our set of universal icons to more complex components of the system.
+description: Icons that are meaningful and understood by all users can improve a user's experience. The icon component is used to apply our set of meaningful icons to more complex components of the system.
```
```suggestion
description: Icons that are meaningful and understood by all users can improve a user’s experience. The icon component is used to apply our set of meaningful icons to more complex components of the system.
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -17,30 +17,33 @@ relatedComponents:
 <DoHeading />
 
-- Only use universal icons, that all users understand, and that clearly represent an intended meaning or context
+- Use icons, that all users understand, and that clearly represent an intended meaning or context.
```
We only have a full stop on the last item
```suggestion
- Use icons, that all users understand, and that clearly represent an intended meaning or context
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -17,30 +17,33 @@ relatedComponents:
 <DoHeading />
 
-- Only use universal icons, that all users understand, and that clearly represent an intended meaning or context
+- Use icons, that all users understand, and that clearly represent an intended meaning or context.
```
We never seem to start these bullet points with capitals elsewhere…

stowball replied:
Still using capitals…

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -17,30 +17,33 @@ relatedComponents:
 - Avoid using bespoke icons in your applications as you may confuse users
-- Only consider using an icon without a meaningful label if there is insufficient space and including the label will detract from the user experience.
-  - An example of this is mobile search bar where including a label for the action will take up essential space for search input, and the icon meaning is explicit like ’search’ or ’close’.
+- Only consider using an icon without a visible label if there is insufficient space and including the label will detract from the user experience.
```
```suggestion
- Only consider using an icon without a visible label where there is insufficient space and including the label will detract from the user experience.
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -17,30 +17,33 @@ relatedComponents:
+## Icons and labels
 
-> Do you have a user need for a new or missing icon? [Get in touch](/about) to discuss an icon contribution.
+Provide a meaningful label and accessible name with the icon to help all users quickly recognise and understand what action will occur. Icons are small visual elements that can be interpreted differently by people. Using icons without labels can confuse some users - even the simplest icons will be new or uncommon to some users.
```
We don't always provide an accessible because often the label is the the thing
```suggestion
Provide a meaningful label and/or accessible name with the icon to help all users quickly recognise and understand what action will occur. Icons are small visual elements that can be interpreted differently by people. Using icons without labels can confuse some users – even the simplest icons will be new or uncommon to some users.
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,68 @@
+<DoHeading />
+
+- Use icons, that all users understand, and that clearly represent an intended meaning or context.
+- Use icons to support visual recognition of repeatable or consistent system wide states or actions
```
```suggestion
- Use icons to support visual recognition of repeatable or consistent system-wide states or actions
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,68 @@
+
+- Use icons, that all users understand, and that clearly represent an intended meaning or context.
+- Use icons to support visual recognition of repeatable or consistent system wide states or actions
+- Always use a meaningful text label and accessible name when using an icon in an action
```
This isn't true. We often don't give an icon an accessible name because it is paired with a meaningful text label

atlee025 replied:
Removed `Always` from dotpoint

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,68 @@
+
+<DontHeading />
+
+- Use icons that are unconventional or are not commonly understood to represent the action or state - this will confuse users.
```
```suggestion
- Use icons that are unconventional or are not commonly understood to represent the action or state – this will confuse users.
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,68 @@
+
+## Icons
+
+Each icon in our library is simple but detailed enough to ensure users can understand them. Icons should not be too detailed, so they are clear and legible at all available sizes. [See the full list of available icons.](/foundations/icons)
```
```suggestion
Each icon in our library is simple, but detailed enough to ensure users can understand them. Icons should not be too detailed, so they are clear and legible at all available sizes. [See the full list of available icons.](/foundations/icons)
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,68 @@
+
+Provide a meaningful label and accessible name with the icon to help all users quickly recognise and understand what action will occur. Icons are small visual elements that can be interpreted differently by people. Using icons without labels can confuse some users - even the simplest icons will be new or uncommon to some users.
+
+An icon can be used without a visible label if there is insufficient space and the context provides clear meaning to the icon. An example of this is a mobile search bar. Including a label for the action will take up essential space needed by the search input, and the context provides clear meaning to the icon.
```
```suggestion
An icon can be used without a visible label when there is insufficient space and the context provides clear meaning to the icon. An example of this is a mobile search bar. Including a label for the action will take up essential space needed by the search input, and the context provides clear meaning to the icon.
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,68 @@
+
+Use icons and labels together that are commonly used and accurately communicate an action, context or outcome. For example, a flag [Toggle button](/components/toggle-button) can be used without a visible label provided there is enough context surrounding the icon, such as in a mailbox application to action its importance or required attention. [Toggle button](/components/toggle-button) will provi ...
+
+## Icon Position
```
```suggestion
## Icon position
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,68 @@
+
+Medium icons are used in small buttons, in components like accordions, drop-down menus, breadcrumbs and links that are nested in content paragraphs. Always consider the complexity of an icon before using the small variant, as small complex icons may be hard to read for some users.
+
+Large icons are most commonly used in medium sized buttons and links. All icons of the system are optimised for legibility at this size.
```
```suggestion
Large icons are most commonly used in medium-sized buttons and links. All icons of the system are optimised for legibility at this size.
```

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,68 @@
+
+## Sizes
+
+Icons are available in sizes medium, large and extra large.
```
```suggestion
Icons are available in three sizes: medium, large and extra large.
```

atlee025 replied:
There was a comment left in the docs to avoid mentioning small sizes: @adrian-yee

atlee025 replied:
Updated to add `small` back into `Sizes`

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,68 @@
+
+Icons are available in sizes medium, large and extra large.
+
+Medium icons are used in small buttons, in components like accordions, drop-down menus, breadcrumbs and links that are nested in content paragraphs. Always consider the complexity of an icon before using the small variant, as small complex icons may be hard to read for some users.
```
This feels weird to tack a thing about small on the end. We should lead with small (so the content increases in size order), states what small is for, but also warns.

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -21,46 +21,47 @@ Meaningful icons can be used to enhance the experience for sighted users; making
+Icons are available in sizes small, medium, large and extra large.
 
-Medium icons are used in small buttons, in components like accordions, drop-down menus, breadcrumbs and links that are nested in content paragraphs. Always consider the complexity of an icon before using the small variant, as small complex icons may be hard to read for some users.
+Small icons are used in small buttons, in components like accordions, drop-down menus, breadcrumbs and links that are nested in content paragraphs. Always consider the complexity of an icon before using the small variant, as small complex icons may be hard to read for some users.
```
This is incorrect. Medium icons are still used in small buttons etc.

My comment was to bring the "Always consider the complexity of an icon before using the small variant, as small complex icons may be hard to read for some users." up first, but instead of just being negative, tell users first what they can be used for, then warn about using them etc

stowball replied:
I see @adrian-yee specifically wanted small to be "not mentioned", but it feels weird and underhanded the way we've done it.

atlee025 replied:
Updated is confirmed good and resolved

atlee025 commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,69 @@
 
 > Do you have a user need for a new or missing icon? [Get in touch](/about) to discuss an icon contribution.
+
+## Icons and labels
+
+Icons are small meaningful images that are used to communicate actions or status. Using icons without labels can confuse some users because even the simplest icons can be misinterpreted. Always provide a meaningful label and/or accessible name with an icon to help all users quickly recognise and understand what has happened or what will occur. Pairing icons with meaningful and common visible labe ...
+
+An icon can be used without a visible label if there is insufficient space and the composition provides clear context and meaning for the icon. For example, in a mailbox or message application a flag [Toggle button](/components/toggle-button) can be used without a visible label to allow a user to mark an item as 'flagged'. Including a label for the action would take up essential space needed by t ...
+
+The [Search box](/components/search-box) component uses a magnifying glass icon with the label ’Search’ in situations where there is enough space for both the icon and label. However, on small screens the [Search box](/components/search-box) responds to a narrower component where only the magnifying glass icon is shown. Keeping both icon and label on the narrow [Search box](/components/search-box ...
```
@stowball Adrian reworded this section, here is the latest

stowball replied:
```suggestion
The [Search box](/components/search-box) component uses a magnifying glass icon with the label ‘Search’ in situations where there is enough space for both the icon and label. However, on small screens the [Search box](/components/search-box) responds to a narrower component where only the magnifying glass icon is shown. Keeping both icon and label on the narrow [Search box](/components/search-box) would reduce the space available for search input which would compromise the utility of the component and the user experience.
```

stowball replied:
This isn't true

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,69 @@
+
+Icons are small meaningful images that are used to communicate actions or status. Using icons without labels can confuse some users because even the simplest icons can be misinterpreted. Always provide a meaningful label and/or accessible name with an icon to help all users quickly recognise and understand what has happened or what will occur. Pairing icons with meaningful and common visible labe ...
+
+An icon can be used without a visible label if there is insufficient space and the composition provides clear context and meaning for the icon. For example, in a mailbox or message application a flag [Toggle button](/components/toggle-button) can be used without a visible label to allow a user to mark an item as 'flagged'. Including a label for the action would take up essential space needed by t ...
```
```suggestion
An icon can be used without a visible label if there is insufficient space and the composition provides clear context and meaning for the icon. For example, in a mailbox or message application a flag [Toggle button](/components/toggle-button) can be used without a visible label to allow a user to mark an item as ‘flagged’. Including a label for the action would take up essential space needed by the table row, and the application provides clear context to the icon meaning. [Toggle button](/components/toggle-button) will provide an accessible label for assistive technologies.
```

stowball replied:
"Including a label for the action would take up essential space needed by the table row, and the application provides clear context to the icon meaning"

I don't like this sentence, specifically the "and…" part, but I can't think of an improvement right now

stowball commented on packages/react/src/icon/docs/overview.mdx:
```diff
@@ -1,14 +1,69 @@
+
+## Icons and labels
+
+Icons are small meaningful images that are used to communicate actions or status. Using icons without labels can confuse some users because even the simplest icons can be misinterpreted. Always provide a meaningful label and/or accessible name with an icon to help all users quickly recognise and understand what has happened or what will occur. Pairing icons with meaningful and common visible labe ...
```
```suggestion
Icons are small meaningful images that are used to communicate actions or status. Using icons without labels can confuse some users because even the simplest icons can be misinterpreted. Always provide a meaningful label and/or accessible name with an icon to help all users quickly recognise and understand what has happened or what will occur. Pairing icons with meaningful and common visible labels helps align with users’ expectations and to communicate an action, context or outcome and provides the best user experience.
```
